mm2src/coins/eth.rs

```diff
@@ -524,6 +525,29 @@ impl EthCoinImpl {
     }
 }
 
+async fn get_raw_transaction_impl(coin: EthCoin, req: RawTransactionRequest) -> RawTransactionResult {
+    let mut tx = req.tx_hash;
+    if tx.len() < 2 {
+        return MmError::err(RawTransactionError::InvalidHashError(tx));
+    }
+    if tx.starts_with("0x") {
+        tx = (tx[2..]).to_string();
+    }
+    let hash = H256::from_str(&tx).map_to_mm(|e| RawTransactionError::InvalidHashError(e.to_string()))?;
+    let web3_tx = coin
+        .web3
+        .eth()
+        .transaction(TransactionId::Hash(hash))
+        .compat()
+        .await
+        .map_err(|e| RawTransactionError::Transport(e.to_string()))?;
```

Comment:
You can implement `From<web3::Error>` for `RawTransactionError` and then use the `?` operator without explicit mapping.
`MmError` allows us to convert any error type `E1` that implements `Into<MmError<E2>>` automatically:
https://github.com/KomodoPlatform/atomicDEX-API/blob/454c571bc493d48930822f41122b5e590593c7f7/mm2src/common/mm_error/mm_error.rs#L148-L156